New Azure function added

PedroRamosProfisee opened the pull request.

PedroRamosProfisee committed:
New Azure function added

PedroRamosProfisee committed:
Fixing function names and appsettings

ZaneJo commented on WebhookTemplate.AzureFunction/Functions/UpdateEntityDescription.cs:
```diff
@@ -0,0 +1,299 @@
+
+public class UpdateEntityDescription
+{
+    [FunctionName("UpdateEntityDescription")]
```
Should this function and file be renamed to make it known that it is meant for the WF activity?

PedroRamosProfisee replied:
Renamed to WorkflowUpdateEntityDescription

ZaneJo commented on WebhookTemplate/appsettings.json:
```diff
@@ -11,6 +11,6 @@
     "LoggingFilePath": "",
     "LoggingConnectionString": "",
-    "ServiceUrl":""
+    "ServiceUrl": "https://corpltr11.corp.profisee.com/Profisee/"
```
Probably should clear this value out.

PedroRamosProfisee replied:
Clearing this value out. We still want an appsettings.json to be filled out but I think we can remove the Profisee portion

ZaneJo commented on WebhookTemplateActivityLibrary/WebhookTemplateActivityLibrary.csproj:
Probably don't want these changes since this is meant to work with the installed SDK of any version, but yours will limit it to 23R2.

PedroRamosProfisee replied:
removed the WebhookTemplateActivityLibrary

ZaneJo commented on WebhookTemplate.AzureFunction/Properties/ServiceDependencies/Story128376FunctionApp - Zip Deploy/appInsights1.arm.json:
Do we need these files? If we do, could we have them renamed so that they don't mention a work item?

PedroRamosProfisee replied:
I'm going to remove these files and remove the publish path for the solution since that's dependent heavily on the user

ZaneJo commented on WebhookTemplate.AzureFunction/Functions/Subscriber.cs:
```diff
@@ -0,0 +1,305 @@
+            Address = config.GetValue<string>("ServiceUrl") + "auth",
+            Policy = { RequireHttps = false, ValidateIssuerName = false }
+        }).Result;
+    }
```
Very minor, but could you add new lines between each of these methods?

PedroRamosProfisee replied:
Lines added

ZaneJo commented on WebhookTemplate.AzureFunction/Functions/Subscriber.cs:
```diff
@@ -0,0 +1,305 @@
+        string msg;
+        try
+        {
+             msg = $"ActivityGeneric HTTP trigger function processed a request.";
```
```suggestion
            msg = $"ActivityGeneric HTTP trigger function processed a request.";
```
I **think** there's an extra space here.

PedroRamosProfisee replied:
Good catch, I removed that extra space

ZaneJo commented on WebhookTemplate.AzureFunction/Functions/Subscriber.cs:
```diff
@@ -0,0 +1,305 @@
+                log.LogInformation($"Header = {item.Key} Value = {item.Value} \n");
+            }
+
+            string authorizationHeader = req.Headers["Authorization"];
```
I may leave it up to James or Lee (or whomever tests this), but do we want to add comments for each section of this method? e.g. Read Request, Validate Token, Update Record, etc.

PedroRamosProfisee replied:
As per Lee's request, I'll go in and add comments to each section that way they can be self-explanatory

ZaneJo commented on WebhookTemplate.AzureFunction/Functions/Subscriber.cs:
```diff
@@ -0,0 +1,305 @@
+            return $"Id: {this.Id}  Name: {this.Name}";
+        }
+    }
+    public class SubscriberResponse
```
The Subscriber function doesn't need to respond with anything, only the WF one does.

PedroRamosProfisee replied:
removed Subscriber response object, method is now a void method and will log any errors

PedroRamosProfisee committed:
Moving up suggested changes from James's review. Testing changes still pending a redeployment of the PaaS system

PedroRamosProfisee committed:
Adding comments to each section

PedroRamosProfisee committed:
Adding sample clock skews

PedroRamosProfisee committed:
Modified .gitignore files to ignore the .vscode file entries

jamesridgwayprofisee committed:
Cleanup

- Cleanup on Azure function classes
- Removed unused usings
- Removed unnecessary properties/methods from Payload folder classes
